lib/mongo/operation/write/update.rb

```diff
@@ -71,12 +76,16 @@ def message(server)
           flags = []
           flags << :multi_update if update[Operation::MULTI]
           flags << :upsert if update[Operation::UPSERT]
+
+          options = flags.empty? ? {} : { flags: flags }
+          options[:array_filters] = update[:array_filters] if has_array_filters?
```

Comment:
You don't need to do any of this, as array filters can only be used with server version 3.6, with which only write commands will be used (not the ``Protocol::Update`` message).